### ISSUE TITLE
goth optimizations (discussion)

### ISSUE THREAD

**Issue by mfranciszkiewicz** [golemfactory/yagna#1531]
1. run agents asynchronously
1. try to setup central net, blockchain and constant components for all tests
1. don't build docker container at startup

**Comment by tworec** [comment]
We can pre-build docker container with all dependencies and at the first test just install proper version of yagna and other our (Golem Factory) artifacts. It should be instant. It will gain ~1 minute. Seems low hanging friut.

**Comment by zakaprov** [comment]
`try to setup central net, blockchain and constant components for all tests`
It's possible (and not too complicated) to keep the "static" network part (i.e. central net, blockchain, proxy) between tests.
However, this would fundamentally change the way we perform tests (we'd no longer have full isolation between test cases).
The time gained would be ~12 seconds per test case.

`run agents asynchronously`
The most time-consuming part of each test right now is waiting for `yagna` nodes to be ready. This takes ~20 seconds **per node**. It's due to how the app keys are currently handled (in order to use the pre-funded Ganache keys we need to perform a restart of the daemon).
I can see two potential improvements here:
1.. Improve how the keys can be loaded (identity pre-configuration or identity drop/update command).
2. Perform node setup in parallel in `goth`.

**Comment by tworec** [comment]
Re1) rather no-go for now
Re2) Seems low hanging fruit, and it needs joit effort from Core and SDK. Cool!

**Comment by nieznanysprawiciel** [comment]
Duplicate of: https://github.com/golemfactory/yagna/issues/1532

**Comment by tworec** [comment]
At our cyclic "make goth great" meeting we've discussed possible goth initialisation optimisation via #1539 
It was added as https://github.com/golemfactory/goth/issues/529
